clients/rabbitmq/src/main/scala/mqperf/rabbitmq/RabbitMq.scala

```
@@ -0,0 +1,133 @@
+
+import java.util.concurrent.ConcurrentLinkedQueue
+import scala.annotation.tailrec
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{Future, blocking}
+import scala.util.Try
+
+class RabbitMq extends Mq with StrictLogging {
+  private val HostsConfigKey = "hosts"
+  private val QueueNameConfigKey = "queueName"
+  private val NrPollReattemptConfigKey = "nrPollReattempt"
+
+  var conn: Connection = _
+
+  override def init(config: Config): Unit = {
+    val host: String = config.mqConfig(HostsConfigKey)
+    val queueName: String = config.mqConfig(QueueNameConfigKey)
+
+    val cf = new ConnectionFactory()
+    cf.setHost(host)
+    cf.setUsername("guest")
+    cf.setPassword("guest")
+
+    try {
+      conn = cf.newConnection()
+
+      newChannel(queueName, passive = false)
+      logger.info(s"Created queue $queueName")
+    } catch {
+      case e: Exception => logger.warn(s"Queue $queueName creation failed", e)
+    }
+  }
+
+  private def newChannel(queueName: String, passive: Boolean): Channel = {
+    val channel = conn.createChannel()
+    if (passive) channel.queueDeclarePassive(queueName)
+    else channel.queueDeclare(queueName, false, false, false, null)
+
+    channel
+  }
+
+  override def createSender(config: Config): MqSender = new MqSender {
+    private val queueName: String = config.mqConfig(QueueNameConfigKey)
+    private val channel = newChannel(queueName, passive = true)
+    channel.confirmSelect()
+
+    override def send(msgs: Seq[String]): Future[Unit] = Future {
+      msgs.foreach(msg => channel.basicPublish("", queueName, MessageProperties.PERSISTENT_TEXT_PLAIN, msg.getBytes))
+    }
+  }
+
+  override def createReceiver(config: Config): MqReceiver = new MqReceiver {
+    override type MsgId = Long
+
+    private val queueName: String = config.mqConfig(QueueNameConfigKey)
+    private val nrPollReattempt: Int = Try(config.mqConfig(NrPollReattemptConfigKey).toInt).getOrElse(0)
```

should we ignore an error though? I'd use Option instead, for example:
```scala
Option(config.mqConfig(NrPollReattemptConfigKey)).map(_.toInt).getOrElse(0)
```